### PR TITLE
implement first pass at serialization

### PR DESCRIPTION
**Pull request by schuyler1d**
Well, where there's deserialization, some folks will need to serialize the other way :-/

Many thanks for your work on deserializing, and extensive test suite.

### CODE REVIEW

File: storable/output.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    return unsigned_int(size, area=area)
+
+
+def signed_smallint(value):
+    """
+    Returns signed value
+    """
+    negative = (value < 0)
+    value = abs(value)
+    assert(value < 128)
```
Comment (exhuma) [inline storable/output.py]: I would personally prefer a `ValueError` here. First, `assert` statements get dropped in "optimized mode" and have no helpful message associated with them. Both issues are resolved by throwing a `ValueError`.

**Review by exhuma — COMMENTED** [review]
Just glanced over it. It looks good. I'll go over it in more detail in a second.

And thanks a lot for the PR 👍

**Comment by exhuma** [comment]
I like these changes! We'll just have to figure out why the tests fail. If we get those fixed, I'll push a new release of the module.

**Comment by schuyler1d** [comment]
@exhuma ok, sorry for the commit-churn, (didn't have all the versions locally, so needed to do a little travis-poking).  now passing tests.

I also added a method I will need, `storable.output.modify_hash` -- it's experimental enough that I'm not sure it's worth documenting, but I added some tests.

**Comment by exhuma** [comment]
Don't worry about the churn. As long as it's working and the tests pass, I'm fine with it ;)

I'll look it through tomorrow and will merge if possible.

**Comment by schuyler1d** [comment]
a friendly ping

**Comment by exhuma** [comment]
Has been merged and pushed to pypi as v1.1.0

Weirdly github did not recognize this as merge, so I'll have to close it manually. If you want you can resync your fork with this one so the "network" graph shows that everything is up-to-date.

I'm a bit afraid that a rebase which I had to do might have borked things up. And yet I was sure it wouldn't 😞 Hope it won't mess things up on your end.

**Comment by schuyler1d** [comment]
Not at, all -- thanks for merging!

**Pull request closed without merging**